pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -566,6 +566,293 @@ func (gce *GCECloud) EnsureTCPLoadBalancerDeleted(name, region string) error {
 	return err
 }
 
+// UrlMap management
+
+// GetUrlMap returns the UrlMap by name.
+func (gce *GCECloud) GetUrlMap(name string) (*compute.UrlMap, error) {
+	return gce.service.UrlMaps.Get(gce.projectID, name).Do()
+}
+
+// CreateUrlMap creates an url map, using the given backend service as the default service.
+func (gce *GCECloud) CreateUrlMap(backend *compute.BackendService, name string) (*compute.UrlMap, error) {
```

Comment:
You mean like Route does right? https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/cloud.go#L119
I didn't want to impose the concept of urlmap/targetproxy/instance groups on all cloudproviders. At this point it'll be boilerplate for the other providers. I'll revist this when I figure out what these concepts map to on the other providers.